packages/office-addin-mock/src/officeMockObject.ts

```diff
@@ -125,11 +135,11 @@ export class OfficeMockObject {
       this.properties.get(scalarPropertyName)?.loadCalled();
       this.makePropertyCallable(scalarPropertyName);
 
-      this.properties
-        .get(scalarPropertyName)
-        ?.properties.forEach((property: OfficeMockObject) => {
-          property.loadCalled();
-        });
+      Array.from(
+        this.properties.get(scalarPropertyName)?.properties.keys() ?? []
+      ).forEach((property: string) => {
+        this.properties.get(scalarPropertyName)?.loadScalar(property);
```

Comment:
It's unclear to me why this change is needed?  It looks like you're replacing a 'loadCalled' method call with a 'loadScalar' call . . . which looks like infinite recursion.

Reply:
It is not infinite because we only call the function recursively when it does have a value. We only call loadScalar on each of the keys, so when the object doesn't have any property, the recurse will end.

LoadScalar does call the `loadCalled` method and it also recurses that is the main difference. The recurse is necessary because of cases where someone might load a navigational property (it is a bad practice, but possible), which causes all properties to be loaded recursively.
I just added a test case to test this scenario.

Reply:
I now did some clean-ups and I think the changes are more readable (just a few line changes). Should be easier to see what's different now. The difference is pretty much `loadCalled` vs recursing with `loadScalar`.